src/create-client.ts

```diff
@@ -35,114 +39,98 @@ const ORGANIZATION_ID_SESSION_STORAGE_KEY = "workos_organization_id";
 
 const REFRESH_LOCK_NAME = "WORKOS_REFRESH_SESSION";
 
-export async function createClient(
-  clientId: string,
-  options: CreateClientOptions = {},
-) {
-  if (!clientId) {
-    throw new NoClientIdProvidedException();
-  }
-
-  const {
-    apiHostname = DEFAULT_HOSTNAME,
-    https = true,
-    port,
-    redirectUri = window.origin,
-    devMode = location.hostname === "localhost" ||
-      location.hostname === "127.0.0.1",
-    // refresh if this is true
-    onBeforeAutoRefresh = () => {
-      return !document.hidden;
-    },
-    onRedirectCallback = (_: RedirectParams) => {},
-    onRefresh: _onRefresh,
-    onRefreshFailure: _onRefreshFailure,
-    refreshBufferInterval = 10,
-  } = options;
-
-  const _clientId = clientId;
-  const _redirectUri = redirectUri;
-  const _baseUrl = `${https ? "https" : "http"}://${apiHostname}${
-    port ? `:${port}` : ""
-  }`;
-  const _useCookie = !devMode;
-  let _authkitClientState: State = "INITIAL";
-
-  const _shouldRefresh = () => {
-    switch (_authkitClientState) {
-      case "INITIAL":
-      case "AUTHENTICATING":
-        return true;
-      case "ERROR":
-        return false;
-      case "AUTHENTICATED":
-        const accessToken = memoryStorage.getItem(storageKeys.accessToken) as
-          | string
-          | undefined;
-        const expiresAt = memoryStorage.getItem(storageKeys.expiresAt) as
-          | number
-          | undefined;
-
-        if (!accessToken || !expiresAt) {
-          return true;
-        }
-
-        const tokenRefreshBufferInSeconds = refreshBufferInterval * 1000;
-        const refreshTime = expiresAt - tokenRefreshBufferInSeconds;
-        return refreshTime < Date.now();
+export class Client {
+  #state: State;
+  #refreshTimer: ReturnType<typeof setTimeout> | undefined;
+
+  readonly #clientId: string;
```

Comment:
Browser support for private methods in JavaScript itself is [pretty good](https://caniuse.com/?search=private%20methods), but we also don't need to worry about it yet since our `tsconfig` targets `es2020`, which means the resulting code doesn't actually make use of this exact syntax yet.

But either way we should be getting pretty good runtime privacy (as opposed to only compile time via the TypeScript `private`), since the compiler generates pretty interesting code. Worth glancing through `dist`.